backend/app/utils/sse.py

```
@@ -0,0 +1,220 @@
+                # Always yield control back to the event loop periodically
+                # This is critical for proper async behavior
+                await asyncio.sleep(0)
+                
+                try:
+                    # Check queue status
+                    queue_size = queue.qsize()
+                    logger.debug(f"Queue status - ID: {queue_id}, Size: {queue_size}")
+                    
+                    if queue_size > 0:
+                        # Queue has items, get next event immediately
+                        logger.debug(f"Queue has items, getting next event immediately")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)